src/utils/generate_structs.jl

```diff
@@ -53,6 +53,16 @@ function {{struct_name}}{T}({{#parameters}}{{^internal}}{{name}}{{#default}}={{d
     {{struct_name}}({{#parameters}}{{^internal}}{{name}}, {{/internal}}{{/parameters}}InfrastructureSystemsInternal())
 end
 {{/parametric}}
+
+{{#defines_ext}}
+function {{struct_name}}({{#parameters}}{{^internal}}{{^ext}}{{^_forecasts}}{{name}}, {{/_forecasts}}{{/ext}}{{/internal}}{{/parameters}}; ext={{#parameters}}{{#ext}}{{default}}{{/ext}}{{/parameters}})
```

Comment:
Why did you have to do something with forecasts?  Seems like the existing logic with 'default' should have taken care of it.

Reply:
You could remove the `defines_ext` block if you define ext before _forecast in the descriptors.

Reply:
I added this to avoid printing the _forecast argument.